Sources/llbuild2/CAS/Object.swift

```diff
@@ -0,0 +1,60 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import Foundation
+
+import NIO
+
+
+public struct LLBCASObject: Equatable {
+    /// The list of references.
+    public let refs: [LLBDataID]
+
+    /// The object data.
+    public let data: ByteBuffer
+
+    public init(refs: [LLBDataID], data: ByteBuffer) {
+        self.refs = refs
+        self.data = data
+    }
+}
+
+public extension LLBCASObject {
+    /// The size of the object data.
+    var size: Int {
+        return data.readableBytes
+    }
+}
+
+// MARK: Codable support for CASObject
+
+extension LLBCASObject: Codable {
+    private enum CodingKeys: String, CodingKey {
+    case refs
```

Comment:
nit: indent?

Reply:
But this is what Xcode wants the indent to be... 😉